src/main/java/com/weirddev/testme/intellij/template/context/MockitoMockBuilder.java

```diff
@@ -226,30 +227,31 @@ public String buildMockArgsMatchers(List<Param> params,String language) {
      */
     @NotNull
     private String deductMatcherTypeMethod(Param param, Language language) {
-        String matcherType;
-        if (param.getType().isVarargs()) {
-            matcherType = "anyVararg";
+        Type type = param.getType();
+        String matcherMethod = resolveMatcherMethod(type);
+        if (language == Language.Scala) {
+            return matcherMethod;
         }
-        else {
-            matcherType = TYPE_TO_ARG_MATCHERS.get(param.getType().getCanonicalName());
-        }
-        if (matcherType == null) {
-            matcherType = "any";
-        }
-        //todo support anyCollection(),anyMap(),anySet() and consider arrays
-        if (language != Language.Scala) {
-            matcherType += "()";
+        else if (!type.isPrimitive() &&  "any".equals(matcherMethod)) {
+            return matcherMethod + "("+ type.getCanonicalName()+(type.isArray()? "[]":"") +".class)";
+        } else {
+            return matcherMethod+"()";
         }
-        return matcherType;
     }
 
-    @SuppressWarnings("unused")
-    @Deprecated
-    public boolean shouldStub(Method testMethod, List<Field> testedClassFields) {
```

Comment:
please keep this method for now. Although its not used. some users might still use it in their custom templates. so it's only marked as @Deprecated for now